File: command/lib/db/compendium/normalization_type.py
from django.db import models


class NormalizationType(models.Model):
    name = models.TextField(unique=True)
    description = models.TextField()
    python_class = models.TextField()

    def to_dict(self):
        fields = ['id', 'name', 'description']
        type_dict = {k: self.__dict__[k] for k in fields}

        return type_dict

File: static/command/js/ext-js/app/model/NormalizationType.js
Ext.define('command.model.NormalizationType', {
    extend: 'Ext.data.Model',
    idProperty: 'id',
    fields: [
        {name: 'id',  type: 'string'},
        {name: 'name',  type: 'string'},
        {name: 'description',  type: 'string'}
    ]
});
